nautobot/core/views/mixins.py

```diff
@@ -473,7 +471,11 @@ def get_queryset(self):
         Override the original `get_queryset()` to apply permission specific to the user and action.
         """
         queryset = super().get_queryset()
-        return queryset.restrict(self.request.user, PERMISSIONS_ACTION_MAP[self.action])
+        return queryset.restrict(self.request.user, self.get_action())
```

Comment:
Might be good to add a unit test verifying the permissions enforcement for `/plugins/example-plugin/example-models/all-names/` (or whatever) is correctly using the custom action permission?